homeassistant/components/switcher_kis/config_flow.py

```diff
@@ -3,16 +3,32 @@
 
 from typing import Any
 
+import voluptuous as vol
+
 from homeassistant import config_entries
 from homeassistant.data_entry_flow import FlowResult
 
-from .const import DATA_DISCOVERY, DOMAIN
+from .const import CONF_TOKEN, DATA_DISCOVERY, DOMAIN
 from .utils import async_discover_devices
 
 
 class SwitcherFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle Switcher config flow."""
 
+    async def _show_setup_form(
+        self, errors: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Show the setup form to the user."""
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(CONF_TOKEN): str,
```

Comment:
Is this really optional?

Reply:
Yes.
This Token is only required for certain devices and above certain devices' firmware